core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -259,15 +260,16 @@ class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging
     if (stopReplicaRequestMap.nonEmpty)
       throw new IllegalStateException("Controller to broker state change requests batch is not empty while creating a " +
         "new one. Some StopReplica state changes %s might be lost ".format(stopReplicaRequestMap.toString()))
-    if (updateMetadataRequestMap.nonEmpty)
+    if (updateMetadataRequestBrokerSet.nonEmpty)
       throw new IllegalStateException("Controller to broker state change requests batch is not empty while creating a " +
-        "new one. Some UpdateMetadata state changes %s might be lost ".format(updateMetadataRequestMap.toString()))
+        "new one. Some UpdateMetadata state changes to brokers %s might be lost ".format(updateMetadataRequestBrokerSet.toString()))
```

Comment:
We can keep that information if it is needed. But I am not sure if we need it. Actually to discussing with Joel, this exception will be thrown only if there is bug in Apache Kafka. At that point we probably don't care what partition is in the metadata.